### PR TITLE
Enable pin control for SWO logging backend

### PR DESCRIPTION
**Pull request by danieldegrasse**
Enable SWO log backend to apply pin control settings, if they are defined for the ARM instrumentation trace macrocell. Add dts bindings for the ARM ITM, and move the `swo-ref-frequency` property to be defined within the ITM node, instead of the `cpu0` node.

This PR also enables SWO logging on the RT1060, which is the platform these changes were tested on.

### CODE REVIEW

File: dts/bindings/debug/arm,armv7m-itm.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Copyright 2022 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: ARMv7 ITM
```
Comment (mbolivar-nordic) [inline dts/bindings/debug/arm,armv7m-itm.yaml:4]: Can you please expand the acronym ITM in these descriptions, so readers have a better chance of figuring out what's going on here?

File: boards/arm/mimxrt1060_evk/doc/index.rst [hunk 2]
```diff
@@ -333,6 +333,12 @@ etc.):
 - Parity: None
 - Stop bits: 1
 
+Using SWO
+---------------------------------
+SWO can be used as a logging backend, by setting ``CONFIG_LOG_BACKEND_SWO=y``.
```
Comment (mbolivar-nordic) [inline boards/arm/mimxrt1060_evk/doc/index.rst:338]: Did you render this HTML and make sure it looks good? It doesn't seem right to me. I would have expected:

```
Using SWO
---------

SWO can be used [...]
```

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]

File: boards/arm/mimxrt1060_evk/doc/index.rst (hunk 2 above)
Comment (danieldegrasse) [reply boards/arm/mimxrt1060_evk/doc/index.rst]: Fixed now, verified the HTML looks good

File: dts/bindings/debug/arm,armv7m-itm.yaml (hunk 1 above)
Comment (danieldegrasse) [reply dts/bindings/debug/arm,armv7m-itm.yaml]: Updated with a more detailed description

**Review by mbolivar-nordic — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

**Comment by carlescufi** [comment]
@danieldegrasse please rebase

**Comment by danieldegrasse** [comment]
@carlescufi Done, thanks

**Review by anangl — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**